near-token-contract/src/lib.rs

```diff
@@ -69,10 +69,15 @@ impl Contract {
     /// executed only if the predecessor account id is the factory.
     ///
     /// Emit `FtMint` event.
+    #[payable]
     pub fn deposit(&mut self, receiver_id: AccountId, amount: U128, memo: Option<String>) {
         // Only the factory can deposit tokens
         self.assert_factory();
 
+        // TODO: Is it ok if the contract auto-registers receiver on deposit?
```

Comment:
This is also related to the idea of separating the create flow from deposit.

To deposit a token you need to have the account registered, but if you are creating the token for the first time then you can't register yourself first (the token does not exist yet). So for now I just had anyone to deposits tokens automatically be registered, but that might not be a good design. It might be better to separate the create flow from deposit, then the contract could be created, the user could register and finally the deposit could be done. This could still make interacting with the Locker more complex though.

Reply:
Yes, let's separate the flow from creation and registration in two different steps.

Reply:
Separated into two steps in #18